tools/pipelines/templates/build-npm-package.yml

```diff
@@ -425,6 +435,87 @@ stages:
               workingDir: ${{ parameters.buildDirectory }}
               customCommand: 'run bundle-analysis:run'
 
+          - ${{ if and(ne(variables['Build.Reason'], 'PullRequest'), ne(variables['System.TeamProject'], 'public')) }}:
+            - task: Bash@3
+              displayName: Create test directory for uploading telemetry
+              inputs:
+                targetType: 'inline'
+                script: |
+                  mkdir ${{ variables.testWorkspace }}
+
+            # Auth to internal feed
+            - task: Bash@3
+              name: Initialize
+              displayName: Initialize .npmrc for telemetry logger
+              inputs:
+                targetType: 'inline'
+                workingDirectory: ${{ variables.testWorkspace }}
+                script: |
+                  echo Initialize package
+                  npm init --yes
+
+                  echo Generating .npmrc
+                  echo "registry=https://registry.npmjs.org" >> ./.npmrc
+                  echo "always-auth=false" >> ./.npmrc
+
+                  echo "@fluidframework:registry=${{ variables.feed }}" >> ./.npmrc
+                  echo "@fluid-experimental:registry=${{ variables.feed }}" >> ./.npmrc
+                  echo "@fluid-internal:registry=${{ variables.devFeed }}" >> ./.npmrc
+                  echo "@ff-internal:registry=https://pkgs.dev.azure.com/fluidframework/internal/_packaging/build/npm/registry/" >> ./.npmrc
+                  echo "@microsoft:registry=https://office.pkgs.visualstudio.com/_packaging/Office/npm/registry/" >> ./.npmrc
```

Comment:
This would be a great use of variable groups.

Reply:
> One thing to be aware of is that pipelines run on the public pipelines don't have access to these variables. I think that should be fine, since our security model should prevent those pipelines from referencing the feeds anyway (they run using credentials associated with the public project rather than the internal one, which is less permissioned).

Yeah, that seems fine.

@Abe27342 I think it's a good idea to also add the URL for the `@microsoft` scope to the variable group. I was about to do it myself but see the description you gave it wouldn't align with that, so running it by you first.

Reply:
yeah, that's fine with me. I adjusted the description accordingly and added an `officeFeed` variable. Also realized I set the other variables incorrectly (links to the ADO feed view rather than the registry) so fixed those :p

Reply:
[ADO item to use the variable group](https://dev.azure.com/fluidframework/internal/_workitems/edit/3266). I'll get to it at some point in the next sprint or the one after unless someone wants to do it urgently and assigns it to themselves 😄 .